internal/sync.go

```diff
@@ -91,16 +91,33 @@ func (s *syncGSuite) SyncUsers() error {
 		ll.Debug("finding user")
 		uu, _ := s.aws.FindUserByEmail(u.PrimaryEmail)
 		if uu != nil {
-			s.users[uu.Username] = uu
+			// check if suspended state is equal or else update the user
+			if uu.Active == u.Suspended {
+				log.Debug("Mismatch active/suspended, updating user")
+				// create new user object and update the user
+				upu, err := s.aws.UpdateUser(aws.UpdateUser(
+					uu.ID,
+					u.Name.GivenName,
+					u.Name.FamilyName,
+					u.PrimaryEmail,
+					!u.Suspended))
+				if err != nil {
+					return err
+				}
+				s.users[uu.Username] = upu
+			} else {
```

Comment:
☝️ you can remove the `else` here, because you anyway override only if `if uu.Active == u.Suspended`.

Reply:
Updated the code.